src/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,251 @@
+import { useRef, useState, ReactNode, FC, ReactElement, KeyboardEventHandler } from 'react';
+import classNames from 'classnames/bind';
+import { Manager, Reference, Popper } from 'react-popper';
+import { useSelect } from 'downshift';
+import { Scrollbars } from 'rc-scrollbars';
+import { useOnClickOutside } from '@common/hooks';
+import { KeyCodes } from '@common/constants/keyCodes';
+import { ReactComponent as ArrowIcon } from '@assets/icons/arrow.svg';
+import { DropdownOption } from './dropdownOption';
+import { DropdownVariant, RenderDropdownOption, DropdownOptionType, DropdownValue } from './types';
+import { OPEN_DROPDOWN_KEY_CODES, CLOSE_DROPDOWN_KEY_CODES } from './constants';
+import { calculateDefaultIndex, calculateNextIndex, calculatePrevIndex } from './utils';
+import styles from './dropdown.module.scss';
+
+const cx = classNames.bind(styles);
+
+interface DropdownProps {
+  options: DropdownOptionType[];
+  value: DropdownValue;
+  disabled?: boolean;
+  error?: string;
+  mobileDisabled?: boolean;
+  title?: string;
+  touched?: boolean;
+  icon?: ReactNode;
+  variant?: DropdownVariant;
+  placeholder?: string;
+  defaultWidth?: boolean;
+  transparentBackground?: boolean;
+  className?: string;
+  toggleButtonClassName?: string;
+  onChange: (value: DropdownValue) => void;
```

Comment:
This onChange handler can be either from Redux-Form, or our custom handler.
And in the case when it comes from the Redux-Form lib, it will has different type.

Reply:
I guess we will fix it during integration with service UI.
For now I suggest to hold this question.